File: Algorithms/cpp/2422. Merge Operations to Turn Array Into a Palindrome/Solution.cpp
// https://leetcode.com/problems/merge-operations-to-turn-array-into-a-palindrome/

class Solution {
public:
    int minimumOperations(vector<int>& nums) {
        if(nums.size() == 1) return 0;
        int l = 0, r = nums.size() - 1, res = 0;
        long lsum = nums[l], rsum = nums[r];
        while(l < r){
            if(lsum == rsum){
                l++;
                r--;
                lsum = nums[l];
                rsum = nums[r];
            }
            else if(lsum < rsum){
                l++;
                lsum += nums[l];
                res++;
            }
            else if(lsum > rsum){
                r--;
                rsum += nums[r];
                res++;
            }
        }
        return res;
    }
};

File: Algorithms/csharp/2422. Merge Operations to Turn Array Into a Palindrome/Solution.cs
// https://leetcode.com/problems/merge-operations-to-turn-array-into-a-palindrome/

public class Solution {
    public int MinimumOperations(int[] nums) {
        if(nums.Length == 1) return 0;
        int l = 0, r = nums.Length - 1, res = 0;
        long lsum = nums[l], rsum = nums[r];
        while(l < r){
            if(lsum == rsum){
                lsum = nums[++l];
                rsum = nums[--r];
            }
            else if(lsum < rsum){
                lsum += nums[++l];
                res++;
            }
            else if(lsum > rsum){
                rsum += nums[--r];
                res++;
            }
        }
        return res;
    }
}